ballerina/tests/xml_to_record_test.bal

```diff
@@ -0,0 +1,566 @@
+                        </address>
+                        <codes>
+                            <item>4</item>
+                            <item>8</item>
+                            <item>9</item>
+                        </codes>
+                    </bookstore>
+                    <!-- some comment -->
+                    <?doc document="book.doc"?>`;
+
+    Commercial expected = {
+        bookstore: {
+            storeName: "foo",
+            postalCode: 94,
+            isOpen: true,
+            address: {
+                street: "Galle Road",
+                city: "Colombo",
+                country: "Sri Lanka"
+            },
+            codes: {
+                item: [4, 8, 9]
+            },
+            _status: "online"
+        }
+    };
+
+    BookStore|Error actual = toRecord(payload);
+    if actual is Error {
+        test:assertFail("failed to convert xml to record: " + actual.message());
+    } else {
+        test:assertEquals(actual, expected, msg = "testToRecordWithAttribues result incorrect");
+    }
+}
+
+type Commercial2 record {
+    BookStore2 bookstore;
+};
+
+type BookStore2 record {
+    string storeName;
+    int postalCode;
+    boolean isOpen;
+    Address3 address;
+    Codes codes;
+    string _status;
+    string _xmlns_ns0;
+};
+
+@test:Config {
+    groups: ["toRecord"]
+}
+isolated function testToRecordWithNamespaces() {
+    xml payload = xml `
+                    <bookstore status="online" xmlns:ns0="http://sample.com/test">
+                        <storeName>foo</storeName>
+                        <postalCode>94</postalCode>
+                        <isOpen>true</isOpen>
+                        <address>
+                            <street>Galle Road</street>
+                            <city>Colombo</city>
+                            <country>Sri Lanka</country>
+                        </address>
+                        <codes>
+                            <item>4</item>
+                            <item>8</item>
+                            <item>9</item>
+                        </codes>
+                    </bookstore>
+                    <!-- some comment -->
+                    <?doc document="book.doc"?>`;
+
+    Commercial2 expected = {
+        bookstore: {
+            storeName: "foo",
+            postalCode: 94,
+            isOpen: true,
+            address: {
+                street: "Galle Road",
+                city: "Colombo",
+                country: "Sri Lanka"
+            },
+            codes: {
+                item: [4, 8, 9]
+            },
+            _xmlns_ns0: "http://sample.com/test",
+            _status: "online"
+        }
+    };
+
+    BookStore|Error actual = toRecord(payload);
+    if actual is Error {
+        test:assertFail("failed to convert xml to record: " + actual.message());
+    } else {
+        test:assertEquals(actual, expected, msg = "testToRecordWithNamespaces result incorrect");
+    }
+}
```

Comment:
Shall we add some negative testcases? where record type is not match with the XML value

Reply:
Added in 5422e3f